Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -650,7 +650,7 @@ namespace AZ
                 {
                     size_t batchStart = batchIndex * batchSize;
                     // If we're in the last batch, we just get the remaining objects
-                    size_t currentBatchCount = batchIndex == batchCount - 1 ? visibleObjectCount % batchSize : batchSize;
+                    size_t currentBatchCount = batchIndex == batchCount - 1 ? visibleObjectCount - batchStart : batchSize;
```

Comment:
Doesn't this break if batchSize is not a multiple of visibleObjectCount? For example if visibleObjectCount = 10 and batchSize = 3 then currentBatchCount will be 7 on the last batch instead of 1. Unless I am mistaken this should work

```
if( (batchIndex < (batchCount - 1)) || ((visibleObjectCount % batchSize) == 0)
{
	currentBatchCount = batchSize;
}
else
{
	currentBatchCount = isibleObjectCount - batchStart;
}
```

Reply:
Please note that in the last iteration of the loop `batchStart` is substracted, not the `batchSize`. 
With `visibleObjectCount = 10` and `batchSize = 3`:
```
batchIndex        | 0 | 1 | 2 | 3
batchStart        | 0 | 3 | 6 | 9
currentBatchCount | 3 | 3 | 3 | 1
```

Reply:
yeah I confused batchStart with batchSize. Ok. Looks good